pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -661,3 +662,27 @@ func (r *ReconcileIBMBlockCSI) getClusterRoleBindings(instance *ibmblockcsi.IBMB
 		nodeSCC,
 	}
 }
+
+func (r *ReconcileIBMBlockCSI) deleteCSIDriver(instance *ibmblockcsi.IBMBlockCSI) error {
+	logger := log.WithName("deleteCSIDriver")
+
+	csiDriver := instance.GenerateCSIDriver()
+	found := &storagev1.CSIDriver{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{
+		Name:      csiDriver.Name,
+		Namespace: csiDriver.Namespace,
+	}, found)
+	if err != nil && errors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		logger.Error(err, "failed to get CSIDriver", "Name", csiDriver.GetName())
+		return err
+	} else {
+		logger.Info("deleting CSIDriver", "Name", csiDriver.GetName())
+		if err := r.client.Delete(context.TODO(), found); err != nil {
+			logger.Error(err, "failed to delete CSIDriver", "Name", csiDriver.GetName())
+			return err
+		}
+	}
```

Comment:
We can simplify this code and remove the double negative:
```suggestion
	if err == nil {
        	logger.Info("deleting CSIDriver", "Name", csiDriver.GetName())
		if err := r.client.Delete(context.TODO(), found); err != nil {
			logger.Error(err, "failed to delete CSIDriver", "Name", csiDriver.GetName())
			return err
		}
	 } else if errors.IsNotFound(err) {
		return nil
	} else {
		logger.Error(err, "failed to get CSIDriver", "Name", csiDriver.GetName())
		return err
	}
```

Reply:
whatever you decide - note that this code structure exists in several places

Reply:
done

Reply:
could you change [said places](https://github.com/IBM/ibm-block-csi-operator/pull/154#discussion_r642520537) while you're at it?